docs/overview.adoc

```diff
@@ -0,0 +1,331 @@
+2. An object representation of the resource is retrieved from the `GetableTypeInterface` instance via `getEntity`, by providing the method with the `id` that was given in the request (i.e. `42`).
+3. A `TransformerAbstract` retrieved via `getTransformer` and the resource type string are used to convert the resource's object representation into the actual response.
+
+Different types of requests require the implementation of different interfaces.
+We can distinguish between the following request types:
+
+* “`get`”, e.g. `GET /article/42`: requires a `GetableTypeInterface` instance
+* “`list`”, e.g. `GET /article`: requires a `ListableTypeInterface` instance
+* “`create`”, e.g. `CREATE /article`: requires a `CreatableTypeInterface` instance
+* “`update`”, e.g. `PATCH /article/42`: requires a `UpdatableTypeInterface` instance
+* “`delete`”, e.g. `DELETE /article/42`: requires a `DeletableTypeInterface` instance
+* accessing resources via requests like `GET /article/42/relationships/comments` has not yet been implemented
+
+It must be noted however, that these interfaces are designed to best fit the needs of the engine, not the developer.
+What this means is that the engine may require some resource-specific task to be done (e.g. fetching data for a resource by its `id`) and is provided with method that fulfills exactly this purpose (e.g. `GetableTypeInterface::getEntity`).
+Meanwhile, the developer is left with the burden of somehow implementing the required `getEntity` method.
+
+While this allows for great flexibility regarding the inner workings of resource type implementations and is reasonably doable for some methods, it gets exceedingly difficult for others.
+Therefore, for most cases it is recommended to extend the `AbstractResourceType`, which is build around the needs of the developer, and using it in conjunction with the `DynamicTransformer` class.
+
+To reiterate: manually implementing the interfaces listed above is only recommended for cases in which the `AbstractResourceType`/`DynamicTransformer` duo is unsuitable for some reason.
+
+=== The `AbstractResourceType` class
+
+It is important to not misunderstand the purpose of the `AbstractResourceType`.
+Even when extending `AbstractResourceType`, all the decisions that are necessary when implementing the interfaces mentioned in the previous section must still be made and the corresponding logic applied.
+However, what this abstract class attempts to do is providing some guidance and encouraging best practices in that process.
+
+To get a better understanding of what all of this means, the following code shows an example in which a specific resource type class was written for `comment` resources, extending from `AbstractResourceType`.
+
+[source,php]
+----
+class CommentType extends AbstractResourceType
```

Comment:
Is "CommentType" the current recommended "naming convention"? 
I read something like "CommentResourceType" before and wonder what do you would prefer/recommend.

Reply:
Added a section regarding the free choice of naming.